### PR TITLE
fix: the placeholder is not hidden when the image is loaded from the cache

### PR DESCRIPTION
**Pull request by u3u**

### CODE REVIEW

File: src/Image.tsx [hunk 1]
```diff
@@ -101,6 +101,12 @@ const ImageInternal: React.FC<ImageProps> = ({
     }
   };
 
+  const getImgRef = (img: HTMLImageElement) => {
+    if (img && img.complete && (img.naturalWidth || img.naturalHeight)) {
```
Comment (shaodahong) [inline src/Image.tsx:105]: `img?.complete` is better

Comment (shaodahong) [reply src/Image.tsx:105]: and maybe need to judge `ImageStatus` if loaded

Comment (u3u) [reply src/Image.tsx:105]: 好的 判断已加载是为了避免一次额外的渲染吗

Comment (shaodahong) [reply src/Image.tsx]: 是的，`getImgRef` 每次 `render` 的时候都会触发一次， status 的判断可以放在前面

Comment (u3u) [reply src/Image.tsx]: 有一行测试没有覆盖到 我该怎么模拟呢

Comment (shaodahong) [reply src/Image.tsx]: Mock `img` 属性试试，用 https://github.com/react-component/util/blob/c2a8ee01bd10eda83e974b1c20de57dd239578ce/src/test/domHook.ts#L7 试试

参考我之前这个 https://github.com/react-component/table/blob/a8b7fdffae2d1831b41ed71e721439b6264cb98a/tests/Sticky.spec.js#L59

Comment (u3u) [reply src/Image.tsx]: 改好了 你再看看

File: tests/placeholder.test.tsx [hunk 2]
```diff
@@ -38,4 +39,37 @@ describe('Placeholder', () => {
 
     expect(wrapper.find('.rc-image-placeholder').get(0)).toBeUndefined();
   });
+
+  it('Cache', () => {
```
Comment (xrkffgg) [inline tests/placeholder.test.tsx:43]: 这个描述可以详细一点

**Review by shaodahong — APPROVED** [review]
LGTM

File: tests/placeholder.test.tsx (hunk 2 above)
Comment (u3u) [reply tests/placeholder.test.tsx]: 我看其他地方的描述都挺简短的 😂

**Comment by shaodahong** [comment]
Thanks

**Comment by HellCatVN** [comment]
@shaodahong I still having problem when image is loaded from cache placeholder is not hidden.
I am using `"antd": "^4.13.1"` and   "name": "rc-image", "version": "5.2.3" is installed with `yarn add antd` and try using rc-image invidual with `yarn add rc-image` not working too

**Comment by shaodahong** [comment]
> I still having problem when image is loaded from cache placeholder is not hidden.
> I am using `"antd": "^4.13.1"` and "name": "rc-image", "version": "5.2.3" is installed with `yarn add antd` and try using rc-image invidual with `yarn add rc-image` not working too

@HellCatVN Thanks for your report, and can you provide a small reproducible example?

**Pull request merged**